GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -754,6 +754,103 @@ private static string RemoveWarnings(string statusString)
             return trimmedStatus;
         }
 
+        public static IReadOnlyList<GitItemStatus> GetLockedFileList(string lockedFilesString)
+        {
+            var lockFiles = new List<GitItemStatus>();
+            const char x = '?';
+
+            var files = lockedFilesString.Split(new[] { '\n' }, StringSplitOptions.RemoveEmptyEntries);
+            for (int n = 0; n < files.Length; n++)
+            {
+                string line = files[n];
+
+                var items = ParseLockedInfo(line);
+
+                string fileName = items[0];
+                string user = items[1];
+                string itemDesc = user.PadRight(20, ' ') + fileName;
+                GitItemStatus gitItemStatus = GitItemStatusFromStatusCharacter(StagedStatus.Unset, itemDesc, x);
+                gitItemStatus.File = fileName;
+                gitItemStatus.User = user;
+
+                lockFiles.Add(gitItemStatus);
+            }
+
+            return lockFiles;
+        }
+
+        private static IReadOnlyList<string> ParseLockedInfo(string line)
```

Comment:
This code is **very** hard to understand. Shouldn't we use the `git lfs locks --json` which is there for this exact purpose of interoperation?

That will also need some unit tests...